packages/eslint-plugin/src/rules/no-unnecessary-type-constraint.ts

```diff
@@ -43,7 +43,7 @@ export default util.createRule({
     function checkRequiresGenericDeclarationDisambiguation(
       filename: string,
     ): boolean {
-      const pathExt = extname(filename).toLocaleLowerCase();
+      const pathExt = extname(filename).toLocaleLowerCase() as ts.Extension;
```

Comment:
iirc it made ci fail, ill undo it and see what happens

Reply:
it doesn't show up locally because you dont have the updated version of the rule, right?
you can tell just from looking at the code that its illegal, because you are using enums + strings together

Reply:
Aha you're right - the build failure was blocking rule update. Thanks!